packages/gcf-utils/src/gcf-utils.ts

```diff
@@ -45,6 +47,59 @@ interface EnqueueTaskParams {
   name: string;
 }
 
+interface LogFn {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  (msg: string, ...args: any[]): void;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  (obj: object, msg?: string, ...args: any[]): void;
+}
+
+/**
+ * A logger standardized logger for Google Cloud Functions
+ */
+export interface GCFLogger {
+  [key: string]: Function;
+  trace: LogFn;
+  debug: LogFn;
+  info: LogFn;
+  warn: LogFn;
+  error: LogFn;
+  metric: LogFn;
+}
+
+let logger: GCFLogger;
+
+/**
+ * Get the singleton instance of GCFLogger
+ */
+export function getLogger(): GCFLogger {
+  if (!logger) {
+    logger = initLogger();
+  }
+  return logger;
+}
+
+export function initLogger(
+  dest?: NodeJS.WritableStream | SonicBoom
+): GCFLogger {
+  const defaultOptions: pino.LoggerOptions = {
+    customLevels: {
+      metric: 30,
+    },
+    level: 'trace',
+  };
+  const defaultDestination = pino.destination({sync: true});
+  const logger = pino(defaultOptions, dest || defaultDestination);
+  return {
```

Comment:
Pino contains a lot more properties than GCFLogger so I'm only returning the properties here relevant to GCFLogger

Reply:
If we can remove the generic string->Function type, we should be able to directly return the Pino logger (as it will match the GCFLogger interface) so we can avoid returning a re-bound object.

Reply:
So actually there's 2 reasons to not remove ``` [key: string]: Function ``` other than it helping with testing...

1. We won't be able to directly return `logger` if we remove the above mapping. This is because `GCFLogger` has function `metric()` which Pino only has as a custom level. We get this error when we remove the mapping and directly return logger: 
```
Property 'metric' is missing in type 'BaseLogger & { [key: string]: LogFn; }' but required in type 'GCFLogger'
```
(`BaseLogger & { [key: string]: LogFn; }` is the interface that Pino implements)

2. In the next step of my implementation I need to add a `flush` function to GCFLogger which will not be a direct binding like the other logging functions are. For `flush` I'll be passing a custom function that checks the destination type and flushes accordingly. So in this case I can't pass the logger directly anyway

Reply:
Ok I've removed `[key:string]: Function` from the `GCFLogger` interface but that doesn't let us return the `logger` as is for the issues mentioned above. 

I've also tried doing:
```typescript
return {...logger, metric: logger.metric.bind(logger)}
```
but that doesn't work either because (due to the way Pino is setup) we need to bind the other functions as well to the logger object. Doing the above satisfies the compiler but results in runtime issues within Pino because the other logging functions like info, trace, etc. don't have the logger object bound to it